src/main/java/org/cyclops/cyclopscore/datastructure/DimPos.java

```diff
@@ -36,29 +42,42 @@ private DimPos(String world, BlockPos blockPos) {
     }
 
     public RegistryKey<World> getWorldKey() {
-        return RegistryKey.getOrCreateKey(Registry.WORLD_KEY, new ResourceLocation(getWorld()));
+        final RegistryKey<World> getKey = worldKeyCache.get(world);
+
+        if (getKey != null) {
+            return getKey;
+        }
+
+        final RegistryKey<World> newKey = RegistryKey.getOrCreateKey(Registry.WORLD_KEY, new ResourceLocation(world));
+        worldKeyCache.put(world, newKey);
+
+        return newKey;
     }
 
     @Nullable
     public World getWorld(boolean forceLoad) {
         if (worldReference == null) {
             if (MinecraftHelpers.isClientSideThread()) {
-                ClientWorld world = Minecraft.getInstance().world;
+                final ClientWorld world = Minecraft.getInstance().world;
+
                 if (world.getDimensionKey().getLocation().toString().equals(this.getWorld())) {
                     this.worldReference = new WeakReference<>(world);
                     return this.worldReference.get();
-                } else {
```

Comment:
Have you tested this yourself?
Because I suspect that this may now crash when run on clients.

Reply:
`if` block always return from method, so `else` statement is useless

Reply:
Added `world != null` check